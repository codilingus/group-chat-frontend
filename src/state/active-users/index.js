import { handleActions, createAction } from 'redux-actions';

const initialState = [0];

const FETCH_ACTIVE_USERS = 'active-users: fetch-active-users';
const FETCH_ACTIVE_USERS_SUCCESS = 'active-users: fetch-active-users-success';
const FETCH_ACTIVE_USERS_FAILURE = 'active-users: fetch-active-users-failure';

export const activeUsersReducer = handleActions({
  [FETCH_ACTIVE_USERS_SUCCESS]: (state, { payload }) => [
    ...payload.activeUsers
  ]
}, initialState);

export const fetchActiveUsers = createAction(FETCH_ACTIVE_USERS);
export const fetchActiveUsersSuccess = createAction(FETCH_ACTIVE_USERS_SUCCESS);
export const fetchActiveUsersFailure = createAction(FETCH_ACTIVE_USERS_FAILURE);
