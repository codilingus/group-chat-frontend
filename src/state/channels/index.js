import { handleActions } from 'redux-actions';

const initialState = [];

export const channelsReducer = handleActions({

}, initialState);
