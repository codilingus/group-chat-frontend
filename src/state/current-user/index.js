import { handleActions, createAction } from 'redux-actions';

const initialState = ({
  isActive: false
});

export const ADD_NEW_USER = 'current-user: add-new-user';
const ADD_NEW_USER_SUCCESS = 'current-user: add-new-user-success';
const ADD_NEW_USER_FAILURE = 'current-user: add-new-user-failure';

export const currentUserReducer = handleActions({
  [ADD_NEW_USER_SUCCESS]: (state) => ({
    ...state,
    isActive: true
  })
}, initialState);

export const addNewUser = createAction(ADD_NEW_USER);
export const addNewUserSuccess = createAction(ADD_NEW_USER_SUCCESS);
export const addNewUserFailure = createAction(ADD_NEW_USER_FAILURE);
