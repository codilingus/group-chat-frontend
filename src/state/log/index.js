import { handleActions, createAction } from 'redux-actions';

const initialState = ({
  loading: false,
  activeUser: null
});

export const LOGIN = 'login: login';
const LOGIN_SUCCESS = 'login: login-success';
const LOGIN_FAILURE = 'login: login-failure';
export const LOGOUT = 'logout: logout';
const LOGOUT_SUCCESS = 'logout: logout-success';
const LOGOUT_FAILURE = 'logout: logout-failure';

export const logReducer = handleActions({
  [LOGIN]: (state) => ({
    ...state,
    loading: true,
    invalidLogin: false,
  }),

  [LOGIN_SUCCESS]: (state, { payload: userDetails }) => ({
    ...state,
    loading: false,
    invalidLogin: false,
    activeUser: userDetails.username
  }),

  [LOGIN_FAILURE]: (state) => ({
    ...state,
    loading: false,
    invalidLogin: true,
  }),

  [LOGOUT]: (state) => ({
    ...state,
    loading: true
  }),

  [LOGOUT_SUCCESS]: (state) => ({
    ...state,
    loading: false,
    activeUser: null
  }),

  [LOGOUT_FAILURE]: (state) => ({
    ...state,
    loading: false
  })
}, initialState);
 


export const logIn = createAction(LOGIN);
export const logInSuccess = createAction(LOGIN_SUCCESS);
export const logInFailure = createAction(LOGIN_FAILURE);
export const logOut = createAction(LOGOUT);
export const logOutSuccess = createAction(LOGOUT_SUCCESS);
export const logOutFailure = createAction(LOGOUT_FAILURE);
