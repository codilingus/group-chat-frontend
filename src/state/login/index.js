import { handleActions, createAction } from 'redux-actions';

const initialState = ({
  loading: false,
  invalidLogin: 'false'
});

export const LOGIN = 'login: login';
const LOGIN_SUCCESS = 'login: login-success';
const LOGIN_FAILURE = 'login: login-failure';

export const loginReducer = handleActions({
  [LOGIN]: (state) => ({
    ...state,
    loading: true,
    invalidLogin: false,
  }),

  [LOGIN_SUCCESS]: (state) => ({
    ...state,
    loading: false,
    invalidLogin: false
  }),

  [LOGIN_FAILURE]: (state) => ({
    ...state,
    loading: false,
    invalidLogin: true
  })
}, initialState);
  
export const logIn = createAction(LOGIN);
export const logInSuccess = createAction(LOGIN_SUCCESS);
export const logInFailure = createAction(LOGIN_FAILURE);
