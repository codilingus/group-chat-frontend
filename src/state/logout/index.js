import { handleActions, createAction } from 'redux-actions';

const initialState = ({
  loading: false
});

export const LOGOUT = 'logout: logout';
const LOGOUT_SUCCESS = 'logout: logout-success';
const LOGOUT_FAILURE = 'logout: logout-failure';

export const loginReducer = handleActions({
  [LOGOUT]: (state) => ({
    ...state,
    loading: true
  }),

  [LOGOUT_SUCCESS]: (state) => ({
    ...state,
    loading: false
  }),

  [LOGOUT_FAILURE]: (state) => ({
    ...state,
    loading: false
  })
}, initialState);

export const logOut = createAction(LOGOUT);
export const logOutSuccess = createAction(LOGOUT_SUCCESS);
export const logOutFailure = createAction(LOGOUT_FAILURE);