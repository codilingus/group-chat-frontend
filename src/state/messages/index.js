import { createAction, handleActions } from 'redux-actions';

const initialState = {
  messages: [
    { text: 'hello', timestamp: 3 },
    { text: 'world', timestamp: 4 }
  ],
  loading: false,
  id: 5,
  conversetionist: ''
};

const CLEAR_MESSAGE_BOARD = 'messages: clear-message-board';
export const FETCH_MESSAGES = 'messages: fetch-messages';
export const FETCH_MESSAGES_SUCCESS = 'messages: fetch-messages-success';
const FETCH_MESSAGES_FAILURE = 'messages: fetch-messages-failure';

export const messagesReducer = handleActions({

  [CLEAR_MESSAGE_BOARD]: (state, { payload }) => ({
    ...state,
    messages: [{}]
  }),

  [FETCH_MESSAGES]: (state) => ({
    ...state,
    loading: true
  }),

  [FETCH_MESSAGES_SUCCESS]: (state, { payload }) => ({
    ...state,
    loading: false,
    messages: [
      ...state.messages,
      ...payload.messages,
    ]
  })

}, initialState);

export const clearMessageBoard = createAction(CLEAR_MESSAGE_BOARD);
export const fetchMessages = createAction(FETCH_MESSAGES);
export const fetchMessagesSuccess = createAction(FETCH_MESSAGES_SUCCESS);
export const fetchMessagesFailure = createAction(FETCH_MESSAGES_FAILURE);
