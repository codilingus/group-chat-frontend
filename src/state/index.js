import { combineReducers } from 'redux';
import { routerReducer } from 'react-router-redux';
import { currentUserReducer } from './current-user/index';
import { channelsReducer } from './channels/index';
import { usersReducer } from './users/index';
import { messagesReducer } from './messages/index';
import { sessionUserReducer } from "./session-user/index";

const rootReducer = combineReducers({
  currentUser: currentUserReducer,
  channels: channelsReducer,
  users: usersReducer,
  messages: messagesReducer,
  sessionUser: sessionUserReducer,
  router: routerReducer 
});

export default rootReducer;
