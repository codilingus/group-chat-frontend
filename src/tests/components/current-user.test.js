import React from 'react';
import { shallow } from 'enzyme';
import configureStore from 'redux-mock-store';
import state from '../fixtures/root-reducer';
import CurrentUser from '../../modules/main/navigator/current-user';

describe('Should render CurrentUser container', () => {
  const mockStore = configureStore();
  let store, wrapper;

  beforeEach(() => {
    store = mockStore(state);
    store.dispatch = jest.fn();
    wrapper = shallow(
      <CurrentUser store={store} />
    );
  });

  test('should render CurrentUser correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should have props', () => {
    expect(wrapper.props())
      .toEqual(expect.objectContaining({
        activeUser: null,
      }));
  });
});
