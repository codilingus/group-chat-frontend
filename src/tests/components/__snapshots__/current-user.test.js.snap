// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render CurrentUser container should render CurrentUser correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Connect(CurrentUser)
    store={
      Object {
        "clearActions": [Function],
        "dispatch": [MockFunction],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      }
    }
  />,

exports[`should render CurrentUser correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <CurrentUser />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "activeUser": null,
      "dispatch": [MockFunction],
      "store": Object {
        "clearActions": [Function],
        "dispatch": [MockFunction],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
      "storeSubscription": Subscription {
        "listeners": Object {
          "clear": [Function],
          "get": [Function],
          "notify": [Function],
          "subscribe": [Function],
        },
        "onStateChange": [Function],
        "parentSub": undefined,
        "store": Object {
          "clearActions": [Function],
          "dispatch": [MockFunction],
          "getActions": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
        },
        "unsubscribe": [Function],
      },
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
    "nodeType": "host",
    "props": Object {
      "children": "Me",
      "className": "current-user-container",
      "onClick": undefined,
    },
    "ref": null,
    "rendered": "Me",
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": "Me",
        "className": "current-user-container",
        "onClick": undefined,
      },
      "ref": null,
      "rendered": "Me",
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
